libbeat/beat/template.go

```diff
@@ -0,0 +1,67 @@
+package beat
+
+/*
+
+Load the index template. This replaces the manual running of
+curl -XPUT 'http://localhost:9200/_template/filebeat?pretty' -d@/etc/filebeat/filebeat.template.json
+
+
+TODO:
+ - Cleanup TODO's
+ - Should the template also be automatically loaded by logstash-input-beats?
+
+*/
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+var template *string
+
+func init() {
+	template = flag.String("template", "", "Path the index template")
+}
+
+func (beat *Beat) loadTemplate() error {
+
+	// TODO: Fin a way to check if flag was even set. If it was set but no path, default path should be used.
```

Comment:
@andrewkroh That's a good idea. I actually have to check if this could be done with one call. As in case lets say 20 beats start up and every beat sends the template, it is quite likely that between a HEAD request and sending the template, and other one already did it.

Also the idea came up to use versioning for the templates.